suite-native/module-settings/src/components/ViewOnly/AutoEjectSwitch.tsx

```diff
@@ -0,0 +1,84 @@
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+    selectIsDeviceAutoEjectEnabled,
+    selectIsNoPhysicalDeviceConnected,
+    toggleAutoEjectThunk,
+} from '@suite-common/wallet-core';
+import { useAlert } from '@suite-native/alerts';
+import { FeatureFlag, useFeatureFlag } from '@suite-native/feature-flags';
+import { Translation } from '@suite-native/intl';
+import { useToast } from '@suite-native/toasts';
+
+import { TouchableSwitchRow, TouchableSwitchRowDescription } from '../TouchableSwitchRow';
+
+export const AutoEjectSwitch = () => {
+    const dispatch = useDispatch();
+
+    const { showAlert, hideAlert } = useAlert();
+
+    const { showToast } = useToast();
+
+    const isNoPhysicalDeviceConnected = useSelector(selectIsNoPhysicalDeviceConnected);
+
+    const isAutoEjectEnabled = useSelector(selectIsDeviceAutoEjectEnabled);
+    const isViewOnlyByDefaultFeatureFlagEnabled = useFeatureFlag(
+        FeatureFlag.IsViewOnlyByDefaultEnabled,
+    );
+
+    if (!isViewOnlyByDefaultFeatureFlagEnabled) return null;
+
+    const onToggleAutoEject = () => {
+        if (!isAutoEjectEnabled) {
+            showToast({
+                variant: 'default',
+                message: isNoPhysicalDeviceConnected ? (
+                    <Translation id="moduleSettings.viewOnly.autoEject.toast.walletsEjected" />
+                ) : (
+                    <Translation id="moduleSettings.viewOnly.autoEject.toast.walletsWillBeEjected" />
+                ),
+            });
+        }
+        dispatch(toggleAutoEjectThunk());
+    };
+
+    const handleToggleAutoEject = () => {
+        if (isAutoEjectEnabled) {
+            onToggleAutoEject();
+        } else {
+            showAlert({
+                title: (
+                    <>
+                        <Translation id="moduleSettings.viewOnly.autoEject.switch.alert.titleNoConnectedTrezor" />
+                        {!isNoPhysicalDeviceConnected && (
+                            <Translation id="moduleSettings.viewOnly.autoEject.switch.alert.titleConnectedTrezor" />
+                        )}
+                    </>
+                ),
+                primaryButtonTitle: (
+                    <Translation id="moduleSettings.viewOnly.autoEject.switch.alert.primaryButtonTitle" />
+                ),
+                primaryButtonVariant: 'redBold',
+                secondaryButtonTitle: <Translation id="generic.buttons.cancel" />,
+                secondaryButtonVariant: 'redElevation0',
+                onPressSecondaryButton: hideAlert,
+                onPressPrimaryButton: onToggleAutoEject,
+            });
+        }
+    };
```

Comment:
I know I know, but line `16-68` could be in a separate useToggleAutoEject hook for readability, just wanted to mention :D Complete NIT

Reply:
I disagree :/ I don't think it adds any value. Hooks should be reusable code and unless there is a reason, I actually think it's counterproductive to just extract it there. 

Sure, if this was some super complicated logic, it might make some sense, but I guess this is really super simple and can stay here. 

Of course, this is just opinion and I'm happy to discuss further, if you want.